### PR TITLE
Added support of blosc2 plugins to the blosc2 filter; Updated c-blosc2 v2.13.2

### LINKED ISSUE #254: Add compilation of blosc2 "plugins"
Following addition of blosc2 (PR #201).

As it is, code in `c-blosc2/plugins` is not built.
Those plugins cannot be used for writing through the DHF5 blosc2 filter, but building those would allow to read chunks written directly with the `blosc2` library.

### PR DESCRIPTION
**Pull request by t20100**
This PR updates the build of the blosc2 filter to add the embedded plugins and support for the blosc2 plugin system.
Still need to add some tests.

closes #254

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -96,8 +96,8 @@ jobs:
           ls dist
 
       - name: Install from source package
-        run:
-          pip install --pre dist/hdf5plugin*
+        run: |
+          pip install --pre "$(ls dist/hdf5plugin-*)[test]" --only-binary blosc2 || pip install --pre dist/hdf5plugin-*
```
Comment (t20100) [inline .github/workflows/ci.yml:100]: blosc2 wheels are not available for python<=3.8, this tries to install blosc2 and retry without it if failed

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -116,8 +116,10 @@ jobs:
         include:
           - python-version: '3.7'
             OLDEST_DEPENDENCIES: 'h5py==2.8.0'
+            EXTRAS: ''
           - python-version: '3.12'
             OLDEST_DEPENDENCIES: 'h5py==3.10.0'
+            EXTRAS: '[test]'
```
Comment (t20100) [inline .github/workflows/release.yml:122]: To install blosc2 for the tests

**Comment by t20100** [comment]
There's an issue with the [release CI](https://github.com/t20100/hdf5plugin/actions/runs/7802372963/job/21279668640) when building `bytedelta.c`:

<details><summary>Full log..</summary>

```
INFO:root:gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DHAVE_PLUGINS=1 -DSHUFFLE_AVX512_ENABLED=1 -DSHUFFLE_NEON_ENABLED=1 -DHAVE_ZLIB=1 -DHAVE_ZSTD=1 -DH5_USE_18_API -DSHUFFLE_SSE2_ENABLED=1 -Isrc/hdf5/include -Isrc/hdf5/include/linux -Isrc/c-blosc2 -Isrc/c-blosc2/blosc -Isrc/c-blosc2/include -Isrc/c-blosc2/plugins/codecs/zfp/include -Isrc/c-blosc2/internal-complibs/lz4-1.9.4 -Isrc/c-blosc/internal-complibs/zlib-1.2.13 -Isrc/c-blosc2/internal-complibs/zstd-1.5.5 -Isrc/c-blosc2/internal-complibs/zstd-1.5.5/common -Isrc/PyTables/hdf5-blosc2/src -I/opt/_internal/cpython-3.11.6/include/python3.11 -c src/c-blosc2/plugins/filters/bytedelta/bytedelta.c -o build/temp.linux-x86_64-cpython-311/src/c-blosc2/plugins/filters/bytedelta/bytedelta.o -O3 -ffast-math -std=gnu99 -pthread -msse2
        In file included from src/c-blosc2/plugins/filters/bytedelta/bytedelta.c:25:
        src/c-blosc2/plugins/filters/bytedelta/bytedelta.c: In function ‘simd_concat’:
        /opt/rh/devtoolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/include/tmmintrin.h:185:1: error: inlining failed in call to ‘always_inline’ ‘_mm_alignr_epi8’: target specific option mismatch
          185 | _mm_alignr_epi8(__m128i __X, __m128i __Y, const int __N)
              | ^~~~~~~~~~~~~~~
        src/c-blosc2/plugins/filters/bytedelta/bytedelta.c:32:54: note: called from here
           32 | bytes16 simd_concat(bytes16 hi, bytes16 lo) { return _mm_alignr_epi8(hi, lo, 15); }
              |                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
        error: command '/opt/rh/devtoolset-10/root/usr/bin/gcc' failed with exit code 1
```

</details>

Due to an older version of gcc? :shrug: since this is manylinux2014.

As opposed to SSE2 and AVX*, there is no compilation option to toggle the use of SSE3.

**Comment by t20100** [comment]
PR https://github.com/Blosc/c-blosc2/pull/586 has been merged, and this fix is included in this PR to make it work until the next release of  `c-blsoc2`.

**Comment by FrancescAlted** [comment]
A [new release](https://github.com/Blosc/c-blosc2/releases/tag/v2.13.2) of C-Blosc2 (2.13.2), with the PR https://github.com/Blosc/c-blosc2/pull/586 in, has been made.  Feel free to check with it.

**Comment by t20100** [comment]
Thanks @FrancescAlted for the quick release! I will update hdf5plugin to use it.

**Comment by t20100** [comment]
If CI is happy, this is ready for review.

**Comment by FrancescAlted** [comment]
> Looks to have an issue in the release CI with a missing symbol:
> 
> ```
> Failed to load filter blosc2: /opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/hdf5plugin/plugins/libh5blosc2.so
> Traceback (most recent call last):
>   File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/hdf5plugin/_utils.py", line 142, in register_filter
>     lib = ctypes.CDLL(filename)
>   File "/opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/ctypes/__init__.py", line 364, in __init__
>     self._handle = _dlopen(self._name, mode)
> OSError: /opt/hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/hdf5plugin/plugins/libh5blosc2.so: undefined symbol: __pow_finite
> ```

Interesting.  I have never seen this myself.  Looks like an issue in clang 10? https://stackoverflow.com/questions/62334452/fast-math-cause-undefined-reference-to-pow-finite

**Comment by t20100** [comment]
I also saw an issue with glibc >=2.31 (used in Ubuntu20.04) having removed some symbols used when compiling with `-ffast-math` compared to previous versions (and here it is compiled with manylinux2014).
I'm now trying with removing `-ffast-math` since it is not used in blosc2 cmake files.

**Comment by t20100** [comment]
Missing symbol issue fixed by removing `-ffast-math` and manually started release CI passes (see https://github.com/t20100/hdf5plugin/actions/runs/7828434787).

**Comment by t20100** [comment]
Ready for me

**Review by vasole — APPROVED** [review]

File: doc/usage.rst
[the remark this answers is not in the document]
Comment (vasole) [reply doc/usage.rst:1]: OK, only if compressed using something else.

**Pull request merged**